plugins/modules/foreman_compute_attribute.py

```diff
@@ -0,0 +1,139 @@
+
+DOCUMENTATION = '''
+---
+module: foreman_compute_attribute
+short_description: Manage Foreman Compute Attributes
+description:
+  - "Manage Foreman Compute Attributes"
+  - "This beta version can create, and update compute attributes"
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+    - "apypie >= 0.0.1"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+  compute_resource:
+    description:
+      - Name of compute resource
+    required: true
+  compute_profile:
+    description:
+      - Name of compute profile
+    required: true
+  vm_attributes:
+    description:
+      - Hash containing the data of vm_attrs
+    required: true
+
+'''
+
+EXAMPLES = '''
+- name: "Create compute attribute"
+  foreman_compute_attribute:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    vm_attrs:
+      memory_mb: '2048'
+      cpu: '2'
+    state: present
+
+- name: "Update compute attribute"
+  foreman_compute_attribute:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    vm_attrs:
+      memory_mb: '1024'
+      cpu: '1'
+    state: present
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'vm_attrs': 'vm_attrs',
+    'compute_profile_id': 'compute_profile_id',
+    'compute_resource_id': 'compute_resource_id',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            compute_profile_name=dict(required=True),
+            compute_resource_name=dict(required=True),
+            compute_profile_id=dict(),
+            compute_resource_id=dict(),
+            vm_attrs=dict(type='dict'),
+        ),
+        supports_check_mode=True,
+    )
+    (entity_dict, state) = module.parse_params()
+
+    module.connect()
+
+    compute_resource = module.find_resource_by_name('compute_resources', name=entity_dict['compute_resource_name'], failsafe=False, thin=False)
+
+    compute_attributes = compute_resource.get('compute_attributes')
+
+    compute_profile = module.find_resource_by_name('compute_profiles', name=entity_dict['compute_profile_name'], failsafe=False, thin=True)
+
+    entity_dict['compute_profile_id'] = compute_profile['id']
+
+    entity_dict['compute_resource_id'] = compute_resource['id']
+
+    entities = filter(lambda item: item.get('compute_profile_id') == compute_profile['id'], compute_attributes)
```

Comment:
For python3, you need to cast this to a list.
Python3 often returns generators, where python2 returns lists.